src/GafferScene/TweakPlug.cpp

```diff
@@ -210,136 +208,77 @@ IECore::MurmurHash TweakPlug::hash() const
 namespace
 {
 
-const char *modeToString( TweakPlug::Mode mode )
-{
-	switch( mode )
-	{
-		case TweakPlug::Replace :
-			return "Replace";
-		case TweakPlug::Add :
-			return "Add";
-		case TweakPlug::Subtract :
-			return "Subtract";
-		case TweakPlug::Multiply :
-			return "Multiply";
-		case TweakPlug::Remove :
-			return "Remove";
-		default :
-			return "Invalid";
-	}
-}
-
 // TODO - if these make sense, I guess they should be pushed back to cortex
 
 // IsColorTypedData
-template< typename T > struct IsColorTypedData : boost::mpl::and_< TypeTraits::IsTypedData<T>, TypeTraits::IsColor< typename TypeTraits::ValueType<T>::type > > {};
+template< typename T > struct IsColorTypedData : boost::mpl::and_< IECore::TypeTraits::IsTypedData<T>, IECore::TypeTraits::IsColor< typename IECore::TypeTraits::ValueType<T>::type > > {};
 
 // SupportsArithmeticData
-template< typename T > struct SupportsArithData : boost::mpl::or_<  TypeTraits::IsNumericSimpleTypedData<T>, TypeTraits::IsVecTypedData<T>, IsColorTypedData<T>> {};
+template< typename T > struct SupportsArithData : boost::mpl::or_<  IECore::TypeTraits::IsNumericSimpleTypedData<T>, IECore::TypeTraits::IsVecTypedData<T>, IsColorTypedData<T>> {};
+
 
 class NumericTweak
 {
 public:
-	NumericTweak( IECore::Data *sourceData, TweakPlug::Mode mode, const std::string &tweakName )
-		: m_sourceData( sourceData ), m_mode( mode ), m_tweakName( tweakName )
+	NumericTweak( const IECore::Data *sourceData, const IECore::Data *tweakData, GafferScene::TweakPlug::Mode mode, const std::string &tweakName )
+		: m_sourceData( sourceData ), m_tweakData( tweakData ), m_mode( mode ), m_tweakName( tweakName )
 	{
 	}
 
 	template<typename T>
 	void operator()( T * data, typename std::enable_if<SupportsArithData<T>::value>::type *enabler = nullptr ) const
 	{
-		T *sourceDataCast = runTimeCast<T>( m_sourceData );
+		const T *sourceDataCast = IECore::runTimeCast<const T>( m_sourceData );
+		const T *tweakDataCast = IECore::runTimeCast<const T>( m_tweakData );
 		switch( m_mode )
 		{
-			case TweakPlug::Add :
-				data->writable() += sourceDataCast->readable();
+			case GafferScene::TweakPlug::Add :
```

Comment:
These extra namespace prefixes are unnecessary now we're back in the `.cpp` file. Would be good to keep the diff to a minimum...

Reply:
Fixed in 9b794ef88ab1ddd6435bc923e50cc3df76017da3